lib/openstack_taster.rb

```diff
@@ -251,37 +254,68 @@ def create_image(instance)
   # @param username [String] the username to use when logging into the instance
   # @return [Boolean] Whether or not the tests succeeded
   def taste_volumes(instance, username)
-    mount_failures = @volumes.reject do |volume|
-      if volume.attachments.any?
-        error_log(instance.logger, 'info', "Volume '#{volume.name}' is already in an attached state; skipping.", true)
-        next
-      end
-
-      unless volume_attach?(instance, volume)
-        error_log(instance.logger, 'error', "Volume '#{volume.name}' failed to attach.", true)
-        next
-      end
-
-      volume_mount_unmount?(instance, username, volume)
+    volume_name = format(
+      '%s-%s-%s',
+      VOLUME_NAME_PREFIX,
+      Time.new.strftime(TIME_SLUG_FORMAT),
+      instance.id
+    )
+    begin
+      volume = @compute_service.volumes.create name: volume_name, size: VOLUME_SIZE, description: VOLUME_DESCRIPTION
+    rescue Excon::Error => e
+      puts 'Failed to create volume. check log for details.'
+      error_log(instance.logger, 'error', e.message)
+      false
+    end
+    volume.reload
```

Comment:
Yes this is required so that it reloads the state of the volume.

Reply:
I mean that the loop right after this line doesn't account for this edge case.

a do ... while loop ([or similar in ruby](https://stackoverflow.com/a/10713963/9085524)) would work better here since it reloads the volume first before testing if it's ready.

Reply:
You have to reload the volume before the loop using the method here:

```ruby
volume.reload
until volume.ready?
  error_log(instance.logger, 'info', "volume #{volume.name} not ready, waiting...", true)
  volume.reload
  sleep 2
end
```

But a more elegant solution wouldn't need to reload volume initially at all:

```ruby
loop do
  volume.reload
  break if volume.ready?
  error_log(instance.logger, 'info', "volume #{volume.name} not ready, waiting...", true)
  sleep 2
end
```